formats/cbor/jvmTest/src/kotlinx/serialization/cbor/CborCompatibilityTest.kt

```diff
@@ -76,4 +79,42 @@ class CborCompatibilityTest {
         val serial = SomeComplexClass.serializer(ListSerializer(Int.serializer()))
         compare(obj, serial)
     }
+
+    @Test
+    fun testFloat() {
+        compare(FloatData(Float.NaN), FloatData.serializer())
+        compare(FloatData(Float.POSITIVE_INFINITY), FloatData.serializer())
+        compare(FloatData(Float.NEGATIVE_INFINITY), FloatData.serializer())
+        compare(FloatData(Float.MAX_VALUE), FloatData.serializer())
+        compare(FloatData(Float.MIN_VALUE), FloatData.serializer())
+        compare(FloatData(0.0f), FloatData.serializer())
+        compare(FloatData(-0.0f), FloatData.serializer())
+        compare(FloatData(-1.0f), FloatData.serializer())
+        compare(FloatData(1.0f), FloatData.serializer())
+        compare(FloatData(123.56f), FloatData.serializer())
+        compare(FloatData(123.0f), FloatData.serializer())
+        // minimal denormalized value in half-precision
+        compare(FloatData(5.9604645E-8f), FloatData.serializer())
+        // maximal denormalized value in half-precision
+        compare(FloatData(0.000060975552f), FloatData.serializer())
+    }
+
+    @Test
+    fun testDouble() {
+        compare(DoubleData(Double.NaN), DoubleData.serializer())
```

Comment:
nit: you can define local function `fun compareDouble(d: Dobule) = compare(DoubleData(d), DoubleData.serializer())` to duplicate less code